src/main/java/com/andersen/hw/config/AppSpringConfig.java

```diff
@@ -1,48 +1,15 @@
 package com.andersen.hw.config;
 
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.PropertySource;
-import org.springframework.jdbc.core.JdbcTemplate;
-import org.springframework.jdbc.datasource.DataSourceTransactionManager;
-import org.springframework.jdbc.datasource.DriverManagerDataSource;
-import org.springframework.transaction.PlatformTransactionManager;
-import org.springframework.transaction.annotation.EnableTransactionManagement;
-
-import javax.sql.DataSource;
 
 @Configuration
-@EnableTransactionManagement
-@ComponentScan(basePackages = "com.andersen.hw")
-@PropertySource("classpath:application.properties")
 public class AppSpringConfig {
-    @Value("${url}")
-    private String URL;
-    @Value("${user}")
-    private String USER;
-    @Value("${password}")
-    private String PASSWORD;
-
-    @Bean
-    public DataSource dataSource() {
-
-        DriverManagerDataSource dataSource = new DriverManagerDataSource();
-        dataSource.setDriverClassName("org.postgresql.Driver");
-        dataSource.setUrl(URL);
-        dataSource.setUsername(USER);
-        dataSource.setPassword(PASSWORD);
-        return dataSource;
-    }
-
-    @Bean
-    public JdbcTemplate jdbcTemplate() {
-        return new JdbcTemplate(dataSource());
-    }
 
     @Bean
-    public PlatformTransactionManager transactionManager() {
-        return new DataSourceTransactionManager(dataSource());
+    @ConditionalOnProperty(name = "custom.bean.enabled", havingValue = "true")
+    public String thisIsMyFirstConditionalBean() {
+        return "My Bean";
     }
```

Comment:
this is not a bean, it is a String.

when you create bean, then you need not just create a method with some condition, you also need to return this been, not a `String`:
```java
    @Bean
    @ConditionalOnProperty(name = "app.allowConditionalBean", havingValue = "true")
    public ThisIsMyFirstConditionalBean thisIsMyFirstConditionalBean() {
        return new ThisIsMyFirstConditionalBean();
    }
```

Reply:
done